schema/deploy/shipping/views.sql

```diff
@@ -68,74 +68,85 @@ drop view if exists shipping.sample_with_best_available_encounter_data_v1;
-
+      select sample_1.sample_id,
+        sample_1.identifier as sample,
+        encounter.site_id,
+        date_or_null(sample_1.details ->> 'date'::text) as collection_date,
+        coalesce(encounter.encountered::date, date_or_null(sample_1.details ->> 'date'::text)) as best_available_encounter_date,
         coalesce(
-          case
-              -- Environmental samples must be processed first, because they're
-              -- often taken at existing human swab sites
-              when manifest.sample_origin = 'es'
-                then 'environmental'
-              else manifest.swab_site
-          end,
-
-          manifest.sample_origin
-        ) as site_manifest_details,
-
-        site_id is not null as has_encounter_data
-
-        from warehouse.sample
+            case
+                when btrim(sample_1.details ->> 'sample_origin'::text, ' '::text) = 'es'::text then 'environmental'::text
+                else btrim(sample_1.details ->> 'swab_site'::text, ' '::text)
+            end, btrim(sample_1.details ->> 'sample_origin'::text, ' '::text)) as site_manifest_details,
+        encounter.site_id is not null as has_encounter_data
+      from warehouse.sample sample_1
         left join warehouse.encounter using (encounter_id)
-        left join specimen_manifest_data as manifest using (sample_id)
-    )
-
-  select
-    sample_id,
-    sample.identifier as sample,
-    has_encounter_data,
-    best_available_encounter_date,
-
-    case
-      when best_available_encounter_date < '2019-10-01'::date then 'Y1'
-      when best_available_encounter_date < '2020-11-01'::date then 'Y2'
-      when best_available_encounter_date < '2021-11-01'::date then 'Y3'
-      else null
-    end as season,
-
-    coalesce(site.site_id, site_details.site_id) as best_available_site_id,
-    coalesce(site.identifier, site_details.site) as best_available_site,
-    coalesce(site.details->>'type', site_type) as best_available_site_type,
-    coalesce(site.details->>'category', site_category) as best_available_site_category
-
-  from warehouse.sample
-  left join samples_with_manifest_data using (sample_id)
-  left join site_details on (site_manifest_details similar to manifest_regex)
-  left join warehouse.site on (samples_with_manifest_data.site_id = site.site_id)
-  ;
+    ),
+    samples_with_site_match as (
+      select samples_with_manifest_data.sample_id,
+        samples_with_manifest_data.sample,
+        samples_with_manifest_data.has_encounter_data,
+        samples_with_manifest_data.best_available_encounter_date,
+        samples_with_manifest_data.site_manifest_details,
+        case
+            when samples_with_manifest_data.best_available_encounter_date < '2019-10-01'::date then 'Y1'::text
+            when samples_with_manifest_data.best_available_encounter_date < '2020-11-01'::date then 'Y2'::text
+            when samples_with_manifest_data.best_available_encounter_date < '2021-11-01'::date then 'Y3'::text
```

Comment:
@davereinhart Would it be worthwhile to add Y4 to the season?

Reply:
Yes, great suggestion! I'll add that now.